idanalysis/eqparams.py

```diff
@@ -0,0 +1,1294 @@
+        """
+        self.ids = list()
+        self._model = _model
+        self._eq_params_nominal = None
+        self._beam = Beam(beam_energy)
+        self.brho = self._beam.brho
+        self.gamma = self._beam.gamma
+        self._emitx = None
+        self._emity = None
+        self._espread = None
+        self._taux = None
+        self._tauy = None
+        self._taue = None
+        self._u0 = None
+
+    @property
+    def model(self):
+        """Sirius model.
+
+        Returns:
+            Pymodels object: Si model
+        """
+        return self._model
+
+    @model.setter
+    def model(self, value):
+        global _model
+        global _mid_subsections
+        global _twiss
+        _model = value
+        mia = pyaccel.lattice.find_indices(_model, "fam_name", "mia")
+        mib = pyaccel.lattice.find_indices(_model, "fam_name", "mib")
+        mip = pyaccel.lattice.find_indices(_model, "fam_name", "mip")
+        _mid_subsections = _np.sort(_np.array(mia + mib + mip))
+        _twiss, *_ = pyaccel.optics.calc_twiss(_model, indices="open")
+        self._model = _model
+
+    @property
+    def eq_params_nominal(self):
+        """Equilibrium parameters from nominal model.
+
+        Returns:
+            EqParamFromRadIntegrals object
+        """
+        return self._eq_params_nominal
+
+    @property
+    def emitx(self):
+        """Cumulative horizontal emittances.
+
+        Returns:
+            numpy array: Horizontal emittance [m rad]
+        """
+        return self._emitx
+
+    @property
+    def emity(self):
+        """Cumulative vertical emittances.
+
+        Returns:
+            numpy array: Vertical emittance [m rad]
+        """
+        return self._emity
+
+    @property
+    def espread(self):
+        """Cumulative energy spread.
+
+        Returns:
+            numpy array: Energy spread
+        """
+        return self._espread
+
+    @property
+    def taux(self):
+        """Cumulative horizontal damping time.
+
+        Returns:
+            numpy array: Horizontal damping time [s]
+        """
+        return self._taux
+
+    @property
+    def tauy(self):
+        """Cumulative vertical damping time.
+
+        Returns:
+            numpy array: Vertical damping time [s]
+        """
+        return self._tauy
+
+    @property
+    def taue(self):
+        """Cumulative longitudinal damping time.
+
+        Returns:
+            numpy array: longitudinal damping time [s]
+        """
+        return self._taue
+
+    @property
+    def u0(self):
+        """Cumulative energy loss per turn.
+
+        Returns:
+            numpy array: Energy loss [keV]
+        """
+        return self._u0
+
+    def get_nominal_model_eqparams(self):
+        """Get nominal model equilibrium parameters.
+
+        Returns:
+            EqParamsFromRadIntegrals object: Object containing Eqparams
+        """
+        self.model = pyaccel.lattice.refine_lattice(
```

Comment:
why are you not using the module constant here?

Reply:
Because we only need a refined lattice in this method, so in the other methods it's not necessary to work with a refined model.